File: src/zetsu/src/errors/database-connection-error.ts
import { CustomError, IErrorStruct } from './custom-error';

export class DatabaseConnectionError extends CustomError {
  statusCode = 500;

  constructor(public reason: string = 'Error connecting to database') {
    super(reason);

    Object.setPrototypeOf(this, DatabaseConnectionError.prototype);
  }

  serializeErrors(): IErrorStruct[] {
    return [{ message: this.reason }];
  }
}

File: src/ashura/errors/database_connection_error.py
from typing import Optional, List
from errors.custom_error import CustomError, IErrorStruct

class DatabaseConnectionError(CustomError):
    status_code = 500
    message = 'Error connecting to database'

    def __init__(self, message: str, field: Optional[str] = None):
        super().__init__(message)
        

    def serialize_errors(self) -> List[IErrorStruct]:
        return [IErrorStruct(message=self.message).__dict__]
